### PR TITLE
Variant theme function fix5.0

### PR DESCRIPTION
**Pull request by bennettclark**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Cherry picked from https://github.com/OfficeDev/office-ui-fabric-react/pull/6047

### CODE REVIEW

**Comment by bennettclark** [comment]
@JasonGore @cliffkoh Can one of you with authorization please merge this?

**Comment by cliffkoh** [comment]
Hi @bennettclark , I tried fixing your perms so you can do the merge. Can you try again? Also, please kindly label these PRs with branch: 5.0 label in the future :)

File: common/changes/@uifabric/variants/variantThemeFunctionFix5.0_2018-08-27-16-17.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@uifabric/variants",
+      "comment": "cherry picked for 5.0",
```
Comment (cliffkoh) [inline common/changes/@uifabric/variants/variantThemeFunctionFix5.0_2018-08-27-16-17.json]: Nit: This cherry pick change file should probably be merged with the original one, i.e. the comment in the other change file should mention `Cherry pick of fixes to....`

Comment (bennettclark) [reply common/changes/@uifabric/variants/variantThemeFunctionFix5.0_2018-08-27-16-17.json]: Ah, I was advised by someone else to make separate changefiles, which was what I was attempting to do here. I'll merge them in the future. Thanks!

Comment (cliffkoh) [reply common/changes/@uifabric/variants/variantThemeFunctionFix5.0_2018-08-27-16-17.json]: The main implication of this is that with 2 changefiles, when this 5.x change is published, you'll see 2 lines in the release notes and it wouldn't really read well/make sense :) Therefore, 1 combined change file is the recommended way so our release notes look clean and makes sense for downstream consumers.

**Pull request merged**